packages/vaadin-virtual-list/src/virtualizer-iron-list-adapter.js

```diff
@@ -133,21 +135,30 @@ export class IronListAdapter {
       return;
     }
 
-    let fvi = this.firstVisibleIndex + this._vidxOffset;
-
-    const fviOffsetBefore = this.__getIndexScrollOffset(fvi);
+    // Record the scroll position before changing the size
+    let fvi; // first visible index
+    let fviOffsetBefore; // scroll offset of the first visible index
+    if (size > 0) {
+      fvi = this.firstVisibleIndex + this._vidxOffset;
+      fviOffsetBefore = this.__getIndexScrollOffset(fvi);
+    }
 
+    // Change the size
     this.__size = size;
     this._itemsChanged({
       path: 'items'
     });
+    flush();
```

Comment:
The additional `flush()` is now required after not invoking `super.scrollToIndex` (which in turn flushed) when the size is 0